docs/xml.md

```diff
@@ -1,6 +1,659 @@
 
-The XML model format has generally been replaced by the newer, more efficient [binary model format](/binary). Some use cases for the XML format still exist, owing to its human readability.
+The XML model format has generally been replaced by the newer, more efficient [binary model format](binary.md). Some use cases for the XML format still exist, owing to its human readability.
 
-This documentation is incomplete. Contributions are welcome.
+## Contents
+- [File Structure](#file-structure)
+- [roblox](#roblox)
+- [Meta](#meta)
+- [External](#external)
+- [Item](#external)
+- [Properties](#properties)
+- [SharedStrings](#sharedstrings)
+	- [SharedString][SharedString-def] (definition)
+- [Type Elements](#type-elements)
+	- [Axes](#axes)
+	- [BinaryString](#binarystring)
+	- [bool](#bool)
+	- [BrickColor](#brickcolor)
+	- [Color3](#color3)
+	- [Color3uint8](#color3uint8)
+	- [ColorSequence](#colorsequence)
+	- [Content](#content)
+	- [CoordinateFrame](#coordinateframe)
+	- [double](#double) (Float64)
+	- [Faces](#faces)
+	- [float](#float) (Float32)
+	- [Font](#font)
+	- [int](#int) (Int32)
+	- [int64](#int64)
+	- [NumberRange](#numberrange)
+	- [NumberSequence](#numbersequence)
+	- [Optional](#optional)
+	- [PhysicalProperties](#physicalproperties)
+	- [ProtectedString](#protectedstring)
+	- [Ray](#ray)
+	- [Rect2D](#rect2d)
+	- [Ref](#ref) (Referent)
+	- [SharedString][SharedString-use] (property type)
+	- [string](#string)
+	- [token](#token) (Enum)
+	- [UDim](#udim)
+	- [UDim2](#udim2)
+	- [UniqueId](#uniqueid)
+	- [Vector2](#vector2)
+	- [Vector3](#vector3)
+	- [Vector3int16](#vector3int16)
+
+## File Structure
+Roblox XML files consist of a single `<roblox>` element, which contain a sequence of other elements. The basic layout of the file structure is as follows:
+
+- Exactly one `<roblox>` element
+	- Zero or more `<Meta>` elements
+	- Zero or more `<External>` elements
+	- One or more `<Item>` elements
+	- One `<Properties>` element
+		- Zero or more type elements
+	- Zero or more `<Item>` elements (this may nest infinitely)
+	- Zero or one `<SharedStrings>` elements
+		- Zero or more `<SharedString>` elements
+
+**To be accepted by Roblox, files MUST start with `<roblox` and end with `</roblox>`, with no whitespace.**
+
+## roblox
+
+This element is the root of the file. There MUST be one `roblox` element in the file.
+
+The following attributes are required for this element:
+
+| Name      | Contents                                                                         |
+|:----------|:---------------------------------------------------------------------------------|
+| `version` | The version of the format this document contains. This MUST be `4` at this time. |
+
+All other attributes are ignored by Roblox, including those defined by Roblox Studio when exporting files.
```

Comment:
Given they're not necessary for the file, I don't think including them in the spec file is a good idea. I would like this to only include information that's relevant for people reading or writing a file.

That said, I am absolutely open to writing the attributes studio generates down *somewhere*. Given there's several elements and an attribute I'm already omitting (they're used internally to save the IDE state), it might make sense to have a supplementary file for data that's useful but not important like this.